storybookjs/storybook
# CLI: fix support for env vars when running `dev`/`build` commands

Closes #21055 

## What I did

Fixed the target of the `getEnvConfig` call to reference an object that is subsequently passed to the `dev`/`build` helper functions. `getEnvConfig` was previously updating `program`, which appears to be a [carryover from the old implementation](https://github.com/storybookjs/storybook/blob/c6209b4db62c48573de9a8a48348c57cfd4fa36f/lib/core-server/src/cli/dev.ts#L76-L84) where options were actually being read from `program`. Since options are now being passed to the `dev`/`build` functions directly, any options derived from environment variables were effectively being ignored.

(It also appears the logic to convert a string port value to a number was also effectively going unused, but as far as I can tell, it didn't matter.)

## How to test

1. Create any sandbox
2. Within the sandbox, run `yarn storybook` with env vars. e.g.,
   - `CI=true yarn storybook` should not open a browser or prompt if there is a port conflict
   - `SBCONFIG_PORT=8888 yarn storybook` should start storybook on port 8888

Not sure a good way to add a test for this behavior?

## Checklist

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
